vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -57,50 +58,112 @@ const (
+		return
+	}
+	for _, tenant := range tenants.Items {
+		// Get all projects for this tenant
+		projects, err := d.client.Tenants.GetProjects(tenant.ID, nil)
+		if err != nil {
+			log.WithFields(log.Fields{"target": d.target}).Warning("Invalid target, unable to get list of projects, exiting.")
+			return
+		}
+		for _, project := range projects.Items {
+			vms, err := d.client.Projects.GetVMs(project.ID, nil)
+			if err != nil {
+				log.WithFields(log.Fields{"target": d.target}).Warning("Invalid target, unable to fetch list of VMs, exiting.")
+				return
+			}
+			for _, vm := range vms.Items {
+				var taskErr error
+				netTask, err := d.client.VMs.GetNetworks(vm.ID)
+				if err == nil {
+					taskErr = d.taskWait(netTask.ID)
+				}
+				if err != nil || taskErr != nil {
+					log.WithFields(log.Fields{"target": d.target, "error": err, "task-error": taskErr}).
+						Warning("Invalid target, unable to get VM properties, exiting.")
+					return
+				}
+				props := netTask.ResourceProperties.(map[string]interface{})
+				networkConn := props["networkConnections"].([]map[string]interface{})
+				for _, network := range networkConn {
+					if hostIP == network["ipAddress"] {
+						d.tenant = tenant.ID
+						d.project = project.ID
+						d.host = vm.ID
+						return
+					}
+				}
+			}
+		}
+	}
+
+	log.WithFields(log.Fields{"target": d.target}).Warning("Unable to identify this host on this target, exiting.")
+	return
 }
 
-func (d *VolumeDriver) verifyTarget() error {
-	// Try fetching the project for the given project ID,
-	// verifies the target (client) and the project.
-	_, err := d.client.Projects.Get(d.project)
+func (d *VolumeDriver) getHostIP() (hostIP string) {
+	ifAddrs, err := net.InterfaceAddrs()
 
-	if err == nil {
-		// Fetch the VM using given host ID.
-		_, err = d.client.VMs.Get(d.hostID)
+	if err != nil {
+		log.WithFields(log.Fields{"error": err}).Warning("Unable to get interface addresses for this host.")
+		return
+	}
+	for _, ifAddr := range ifAddrs {
+		ipaddr := ifAddr.(*net.IPNet)
+		if !ipaddr.IP.IsLoopback() && ipaddr.IP.To4() != nil {
+			hostIP = ipaddr.IP.String()
+			log.WithFields(log.Fields{"host IP": hostIP}).Info("Found host IP - ")
+			return
+		}
 	}
-	return err
+	return
 }
 
 // NewVolumeDriver - creates Driver, creates client for given target
-func NewVolumeDriver(targetURL string, projectID string, hostID string, mountDir string) *VolumeDriver {
+func NewVolumeDriver(targetURL string, mountDir string) *VolumeDriver {
 
 	d := &VolumeDriver{
-		target:  targetURL,
-		project: projectID,
-		hostID:  hostID,
+		target: targetURL,
 	}
+
 	// Use default timeout of thirty seconds and retry of three
 	d.client = photon.NewClient(targetURL, nil, nil)
 
-	err := d.verifyTarget()
-	if err != nil {
-		log.WithFields(log.Fields{"target": targetURL, "project-id": projectID}).Warning("Invalid target and or project ID, exiting.")
+	// Get host IP address
+	hostIP := d.getHostIP()
```

Comment:
It seems this hostIP can be an empty string if there is no valid IPv4 address on this node. Should we add a check and stop here if it's empty? because it won't match in identifyHost function. And even it matches, it could be the wrong one.